pkg/collector/providers/providers.go

```diff
@@ -21,6 +21,20 @@ func RegisterProvider(name string, factory ConfigProviderFactory) {
 // ConfigProviderFactory is any function capable to create a ConfigProvider instance
 type ConfigProviderFactory func(cfg config.ConfigurationProviders) (ConfigProvider, error)
 
+// CacheProviderIndx contains the number of AD Templates and the latest Index
+type CacheProviderIndx struct {
+	AdTemplate2Idx float64
```

Comment:
The name is meh, we can skip `Ad` since this is in the provider package anyway, and it's not a mapping since we just have the latest. What do you think about `LatestTemplateIdx`?